gui/src/app/SamplingOptsPanel/SamplingOptsPanel.tsx

```diff
@@ -0,0 +1,166 @@
+import { FunctionComponent, useCallback } from "react";
+import { SamplingOpts, defaultSamplingOpts } from "../StanSampler/StanSampler";
+import { Hyperlink } from "@fi-sci/misc";
+
+type SamplingOptsPanelProps = {
+    samplingOpts: SamplingOpts;
+    setSamplingOpts?: (opts: SamplingOpts) => void;
+}
+
+// From Brian
+// The following would be nice to have control over:
+
+// [ ]Number of chains (default 4, in practice I think this being a slider from 1 to 8 is probably reasonable, but in theory it can be unbounded)
+//  Number of warmup iterations (default 1000, can be [0,
+// ))
+//  Number of sampling iterations (default 1000, can be [1,
+// ))
+//  Initializations. This is tricky, can either be 1 json object or a list of num_chains JSON objects.
+//  "Initialization radius" (parameters not given an initial value directly are drawn from uniform(-R, R) on the unconstrained scale) (default 2.0, can be [0,
+// ), but in practice limiting to say 10 in the UI is probably fine)
+//  Seed. Any uint32.
+
+
+const SamplingOptsPanel: FunctionComponent<SamplingOptsPanelProps> = ({ samplingOpts, setSamplingOpts }) => {
+    const num_chains = samplingOpts.num_chains;
+    const readOnly = !setSamplingOpts;
+    const handleReset = useCallback(() => {
+        setSamplingOpts && setSamplingOpts(defaultSamplingOpts)
+    }, [setSamplingOpts])
+    return (
+        <div>
+            <table>
```

Comment:
Not fussed about it right now, but it'd probably be better not to use `table` for layout.

Reply:
I think a table might be okay here. But you are welcome to take a crack at a different approach.

Reply:
I think eventually a `Grid` from materialUI would be nice, since it lets you do some responsive sizing relatively easily. 

We probably never expect someone to use this on a phone, but maybe an ipad is reasonable if we get our layout right